### DISCUSSION TITLE
Ordering in v3

### DISCUSSION THREAD

**Discussion by alexeygolev** [sanity-io/sanity#3713] [Sanity Studio Support] [answered]
Is ordering broken in v3? The only ordering that seem to be working is by dateCreated. Neither out-of-the-box `title`, `name` field sorting nor custom ordering seem to work. Sometimes switching the ordering “hangs” the list to the point that no documents are ever loaded.

I also checked the `viewOptions` argument to the `prepare` function in the preview object and whichever option is selected the value is `undefined`.

I looked at the queries that the studio uses and the order is always by `_updatedAt desc` no matter what ordering is selected. Switching to another ordering doesn’t change the ordering but does add the field the ordering is supposed to use to the list of fields requested on a document type.

**Comment by geball** [comment]
To clarify, are you referring to setting a default ordering (`S.documentList().defaultOrdering()`) or ordering via the three-dot contextual menu (`S.documentList().menuItems()`)? [Edit: I suspect it's the latter, and after testing I'm getting the same "hanging" list of documents that never stops loading.]

**Comment by alexeygolev** [comment]
@geball The default ordering actually works. It's the contextual menu. "Hanging" is just one part of the problem. The ordering itself is not really correct. From what I can see it's always `_updatedAt desc`. At least in my studio even on simple document types with a title field. It infers the `title` field and correctly adds it to the menu. But ordering by it doesn't yield the correct results

**Comment by nilsborg** [comment]
We have exactly the same issue. Even when i add an explicit `ordering` in the schema, it does get picked up by the context menu but doesn't do anything .. document last edited still jump to the top and the rest seems random.

**Reply by nilsborg** [reply]
Any update on this? Seems to be still broken in `preview.22` and it's possibly an easy fix...? More complex things like search and reference sorting got fixed but sort order not 😅

**Reply by nilsborg** [reply]
@alexeygolev did you find a solution by any chance?

**Comment by mstapp** [comment]
Still broken in 3.0.0-rc.0. Any update on this?

**Reply by vicmeow** [reply]
Hi, we have looked into it and believe to have found the issue. The fix will be out in the next release once approved.

**Comment by mstapp** [comment, marked as answer]
It looks like @vicmeow fixed this & the fix was just released in studio 3.1 -- thank you! (Housekeeping info: [PR #3897](https://github.com/sanity-io/sanity/pull/3897) seems to be the one that fixed this issue.)
